packages/askar/src/wallet/AskarWallet.ts

```diff
@@ -1195,14 +1286,69 @@ export class AskarWallet implements Wallet {
           }
         } finally {
           cek?.handle.free()
-          recipientAskarKey?.key?.handle.free()
+          recipientAskarKey?.handle.free()
         }
       }
     } finally {
       senderAskarKey?.handle.free()
       epk?.handle.free()
     }
-    throw new AriesFrameworkError('Unable to decrypt didcomm v2 envelop')
+    throw new AriesFrameworkError('Unable to open jwe: recipient key not found in the wallet')
+  }
+
+  public async resolveRecipientKey({
+    kid,
+    recipientDidDocuments,
+  }: {
+    kid: string
+    recipientDidDocuments: DidDocument[]
+  }): Promise<{ recipientKey: Key | undefined; recipientAskarKey: AskarKey | undefined }> {
+    const recipientDidDocument = recipientDidDocuments.find((didDocument) => keyReferenceToKey(didDocument, kid))
+    if (!recipientDidDocument) {
+      throw new AriesFrameworkError(`Unable to resolve recipient Did Document for kid: ${kid}`)
+    }
+    const recipientKey = keyReferenceToKey(recipientDidDocument, kid)
+    const recipientAskarKey = await this.session.fetchKey({
+      name: recipientKey.publicKeyBase58,
+    })
+    return {
+      recipientKey,
+      recipientAskarKey: recipientAskarKey?.key,
+    }
+  }
+
+  public resolveSenderKeys({
```

Comment:
these can be private right?

Reply:
I initially made it public for mocking in tests, but now tests can be written without using mock.